### ISSUE TITLE
How to run test_bench.py with ROCM?

### ISSUE THREAD

**Issue by jinsong-mao** [pytorch/benchmark#2040]
Hi @xuzhao9, 

I don't know how to create a dockerfile for AMD ROCM,  is there any example?  

Best Regards

**Comment by xuzhao9** [comment]
You could use the userbenchmark rocm_test: https://github.com/pytorch/benchmark/blob/main/userbenchmark/rocm-test/run.py

**Comment by jinsong-mao** [comment]
Hi @xuzhao9  thanks for you remind,  I can reproduce the userbenchmark command for llama on cuda like this:
![image](https://github.com/pytorch/benchmark/assets/146043398/9b0735ed-8825-4cef-95be-0201d6d25060)

1, the output metric is very simple, it doesn't give me any information like inference token per second, input/output sequence length, batch size(default 32),  how should I get that information?

2, I also wants to test different composition of input/output sequence length and even batch size of llama test, what should I do to achieve this?

3, the training test for llama looks failed with some issues, is it my fault or it's the current status of llama?
![image](https://github.com/pytorch/benchmark/assets/146043398/9256b3f7-d290-49a1-8ac6-438ef890b204)

many thanks for your kindly feedback. 
Best

**Comment by xuzhao9** [comment]
@jinsong-mao You could read the README.md file at https://github.com/pytorch/pytorch/tree/main/benchmarks/dynamo for more details. The `dynamo` userbenchmark is owned by the PyTorch 2.0 team, so my answers could be inaccurate, but here are my takes:

1. The input/output sequence length is hard-coded and not tunable: https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/llama/__init__.py#L21. It is possible to make it a model argument, but that would need additional work. Based on my understanding, the input are tensors, not tokens, so it cannot provide information like tokens per second.
2. Since the seqlen is hardcoded right now, you can't test different seqlens. However, different batch sizes are tunable with `--batch-size` (https://github.com/pytorch/pytorch/blob/main/benchmarks/dynamo/common.py#L2905)
3. Train test is not implemented right now. We welcome contribution to enable it.
